src/matio.h

```diff
@@ -230,95 +230,95 @@ typedef struct mat_sparse_t {
 /** @endcond */
 
 /* Library function */
-EXTERN void Mat_GetLibraryVersion(int *major,int *minor,int *release);
+EXTERN MATIO_EXPORT void Mat_GetLibraryVersion(int *major,int *minor,int *release);
```

Comment:
Please try to be minimal invasive on existing files. EXTERN alone should be enough as it can also handle the export definition.

Reply:
EXTERN will also export all the other routines defined in mat4.c mat5.c and mat7.3.c. So it's not appropriate I think. The routines marked with MATIO_EXPORT are those mentioned in matio.def... Maybe the best way to fuse both autoconf and cmake approaches would be to use the cmake way of marking exported routines with this MATIO_EXPORT macro. This has the nice advantage that visibility is directly controlled in one place (and explicitly visible to the user). I do not know how to change the VisualStudio project files to do that, but I can produce the proper matio_export.h from autoconf if I can convince you this is a better route to follow... ;-)

After seeing your comment, I saw a way to add the matio.def file in cmake. But honestly, nowadays marking the routines in the source file seems to be somewhat preferred (all the more that it adds visibility control for gcc which does not have the .def equivalent).

Reply:
Hm, in 001704f2425c0e86c3139c6c5651f0f2a9286667 I added the matio.sym file to fix the visibility control for gcc.